File: matrix-gradle-plugin/src/main/java/io/matrix/gradle/dependencymanagement/MatrixPlugin.java
package io.matrix.gradle.dependencymanagement;

import io.spring.gradle.dependencymanagement.DependencyManagementPlugin;
import io.spring.gradle.dependencymanagement.dsl.DependencyManagementExtension;
import org.gradle.api.Plugin;
import org.gradle.api.Project;
import org.springframework.boot.gradle.plugin.SpringBootPlugin;
import org.springframework.boot.gradle.util.VersionExtractor;


/**
 * @author Noa Swartz
 * @date 2020/09/25
 */
public class MatrixPlugin implements Plugin<Project> {

    static final String BOM_COORDINATES = "io.matrix:matrix-parent:";

    @Override
    public void apply(final Project project) {
        project.getPlugins().apply(DependencyManagementPlugin.class);
        project.getPlugins().apply(SpringBootPlugin.class);
        String version = VersionExtractor.forClass(MatrixPlugin.class);
        project.getExtensions().findByType(DependencyManagementExtension.class)
                .imports((importsHandler) -> importsHandler.mavenBom(MatrixPlugin.BOM_COORDINATES + version));
    }

}

File: matrix-gradle-plugin/src/main/kotlin/io/matrix/gradle/dependencymanagement/MatrixPlugin.kt
package io.matrix.gradle.dependencymanagement

import org.gradle.api.Plugin
import org.gradle.api.Project
import io.spring.gradle.dependencymanagement.DependencyManagementPlugin
import org.springframework.boot.gradle.plugin.SpringBootPlugin

/**
 *
 * @author Noa Swartz
 * @date 2020/09/05
 */
class MatrixPlugin : Plugin<Project> {

    override fun apply(project: Project) {
        SpringBootPlugin().apply(project)
        DependencyManagementPlugin().apply(project)
    }

}